Bug 708606: Catch a null file pointer closing pdfwrite.
In the event of an error opening a new output file.

--- ghostscript-10.05.1/devices/vector/gdevpdf.c
+++ ghostscript-10.05.1/devices/vector/gdevpdf.c
@@ -961,7 +961,10 @@ pdf_ferror(gx_device_pdf *pdev)
 {
     int code = 0;
 
-    gp_fflush(pdev->file);
+    if (pdev->file != NULL) {
+        gp_fflush(pdev->file);
+        code = gp_ferror(pdev->file);
+    }
     gp_fflush(pdev->xref.file);
     if (pdev->strm->file != NULL)
         sflush(pdev->strm);
@@ -970,12 +973,13 @@ sflush(pdev->asides.strm);
     if (pdev->streams.strm->file != NULL)
         sflush(pdev->streams.strm);
     if (pdev->ObjStm.strm != NULL && pdev->ObjStm.strm->file != NULL) {
+        int code2;
         sflush(pdev->ObjStm.strm);
-        code = gp_ferror(pdev->ObjStm.file);
+        code2 = gp_ferror(pdev->ObjStm.file);
+        if (code >= 0) code = code2;
     }
-    return gp_ferror(pdev->file) || gp_ferror(pdev->xref.file) ||
-        gp_ferror(pdev->asides.file) || gp_ferror(pdev->streams.file) ||
-        code;
+    return gp_ferror(pdev->xref.file) || gp_ferror(pdev->asides.file) ||
+           gp_ferror(pdev->streams.file) || code;
 }
 
 /* Compute the dominant text orientation of a page. */
